torch/_dynamo/symbolic_convert.py

```diff
@@ -599,6 +602,23 @@ def inline_user_function_return(self, fn, args, kwargs):
             self.restore_graphstate(state)
             raise
 
+    def log_starts_line(self):
+        inline_depth_str = (
+            f"(inline depth: {self.inline_depth}) " if self.inline_depth > 0 else ""
+        )
+        starts_line_log.debug(
+            "TRACE starts_line %s%s %s:%s",
+            inline_depth_str,
+            self.f_code.co_name,
+            self.f_code.co_filename,
+            self.lineno,
+        )
+        line = linecache.getline(self.f_code.co_filename, self.lineno)
+        if line:
+            # strip the newline at the end
+            line = line[:-1]
+        starts_line_log.debug("    %s", (" " * 4 * self.inline_depth) + line)
```

Comment:
As written, this always does the string formatting even if the debug log is ignored. The formatting here is pretty complicated, maybe best to test log level and short circuit out.